### ISSUE TITLE
Revive this feedstock

### ISSUE THREAD

**Issue by agriyakhetarpal** [conda-forge/pybamm-feedstock#65]
This is a medium-term issue that we should start considering to do since PyBaMM will migrate to using the IDAKLU solver as the default instead of the CasadiSolver.

This requires fixing the sdist, compiling SuiteSparse, and compiling SUNDIALS with KLU linked from SuiteSparse. We might need to patch PyBaMM to not use the CasADi Python package and compile the C interface ourselves (with minimal interfaces), and later turn binary relocation on for the feedstock. We could also introduce additional extension modules for this task.

We can then remove it and package `pybammsolvers` as a separate conda-forge feedstock to use here as a dependency once i. https://github.com/pybamm-team/PyBaMM/pull/4487 is done and ii. things are stable enough for at least one release after the transition

**Comment by agriyakhetarpal** [comment]
This is where I begin to feel that migrating to Meson instead of scikit-build-core might have been better – no one wants to deal with CMake 😃

**Comment by martinjrobins** [comment]
> This is where I begin to feel that migrating to Meson instead of scikit-build-core might have been better – no one wants to deal with CMake 😃

But I've already spent too much of my life fighting with CMake and learning its weird eccentricities!

**Comment by agriyakhetarpal** [comment]
@prady0rt, how is this going? Do you need any help somewhere? Please let us know if you do!

**Comment by agriyakhetarpal** [comment]
Hi @prady0t, how is this going – do you need any help somewhere? Please let us know if you do!

**Comment by prady0t** [comment]
> Hi @prady0t, how is this going – do you need any help somewhere? Please let us know if you do!

Few exam days before I can get back to this 🤒

**Comment by kratman** [comment]
Just to confirm, we don't need to package suitesparse and sundials in the feedstock anymore since we have pybammsolvers correct? Just add pybamm solvers to conda-forge instead?

**Comment by Saransh-cpp** [comment]
Yes, pybammsolvers should be a separate (and a new) conda-forge package. pybamm should just include it as a dependency. I am not sure how the compiling SuiteSparse + SUNDIALS with KLU linked from SuiteSparse thing will go, but it definitely will be required in the new feedstock.

@prady0t could you create a new feedstock request PR so that we all can monitor how things are working or where they are failing?

**Comment by agriyakhetarpal** [comment]
Yes, it still has to be compiled, just elsewhere now. Let's leave this open until said PR is merged.

**Comment by agriyakhetarpal** [comment]
Yes, it still has to be compiled, just elsewhere now. Let's leave this open until the feedstock is set up.

**Comment by Saransh-cpp** [comment]
@conda-forge-admin, please update version

<img width="300" height="167" alt="Image" src="https://github.com/user-attachments/assets/ce48e760-706b-4877-a8fa-dd4388f697c3" />

Edit: the latest version still uses `<0.7.0`:

https://github.com/pybamm-team/PyBaMM/blob/7aa106e66f99a72548a607b85e8874de8747b6a3/pyproject.toml#L30

So need to wait more :(

**Comment by Saransh-cpp** [comment]
@conda-forge-admin, please update version

**Comment by Saransh-cpp** [comment]
Both pybammsolvers and pybamm are up on conda-forge
